smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,677 @@
+
+    metadata = persistable_entity["telemetry_metadata"]
+    status_dir = pathlib.Path(metadata.get("status_dir"))
+
+    entity.type = entity_type
+    entity.name = persistable_entity["name"]
+    entity.job_id = str(metadata.get("job_id", ""))
+    entity.step_id = str(metadata.get("step_id", ""))
+    entity.timestamp = int(persistable_entity.get("timestamp", "0"))
+    entity.path = str(exp_dir)
+    entity.status_dir = str(status_dir)
+
+    entities.append(entity)
+
+    return entities
+
+
+def hydrate_persistables(
+    entity_type: _ManifestKey,
+    run: t.Dict[_ManifestKey, t.Any],
+    exp_dir: pathlib.Path,
+) -> t.List[JobEntity]:
+    """Map a collection of entity data persisted in a manifest file to an object"""
+    persisted: t.List[JobEntity] = []
+
+    for item in run[entity_type]:
+        entities = hydrate_persistable(entity_type, item, exp_dir)
+        for entity in entities:
+            persisted.append(entity)
+
+    return persisted
+
+
+def hydrate_runs(
+    persisted_runs: t.List[t.Dict[_ManifestKey, t.Any]], exp_dir: pathlib.Path
+) -> t.List[Run]:
+    """Map run data persisted in a manifest file to an object"""
+    runs = [
+        Run(
+            timestamp=instance["timestamp"],
+            models=hydrate_persistables("model", instance, exp_dir),
+            orchestrators=hydrate_persistables("orchestrator", instance, exp_dir),
+            ensembles=hydrate_persistables("ensemble", instance, exp_dir),
+        )
+        for instance in persisted_runs
+    ]
+    return runs
+
+
+def load_manifest(file_path: str) -> t.Optional[RuntimeManifest]:
+    """Load a persisted manifest and return the content"""
+    source = pathlib.Path(file_path)
+    source = source.resolve()
+
+    text = source.read_text(encoding="utf-8")
+    text = text.strip()
+    if not text:
+        return None
+
+    manifest_dict = json.loads(text)
+    exp = manifest_dict.get("experiment", None)
+    if not exp:
+        raise ValueError("Manifest missing required experiment")
+
+    runs = manifest_dict.get("runs", None)
+    if runs is None:
+        raise ValueError("Manifest missing required runs")
+
+    exp_dir = pathlib.Path(exp["path"])
+    runs = hydrate_runs(runs, exp_dir)
+
+    manifest = RuntimeManifest(
+        name=exp["name"],
+        path=exp_dir,
+        launcher=exp["launcher"],
+        runs=runs,
+    )
+    return manifest
+
+
+def track_event(
+    timestamp: int,
+    ename: str,
+    job_id: str,
+    step_id: str,
+    etype: str,
+    action: _EventClass,
+    exp_dir: pathlib.Path,
+    logger: logging.Logger,
+    detail: str = "",
+    return_code: t.Optional[int] = None,
+) -> None:
+    """
+    Persist a tracking event for an entity
+    """
+    job_id = job_id or ""
+    step_id = step_id or ""
+    entity_type = etype or "missing_entity_type"
+
+    name: str = ename or "entity-name-not-found"
+    tgt_path = exp_dir / TELMON_SUBDIR / entity_type / name / f"{action}.json"
```

Comment:
Where are shard statuses currently being written to? I thought the path was something like:
`exp_dir/.smartsim/telemetry/<run_id>/<entity_type>/<orc_name> /<shard_id>/start.json `

Reply:
leftover from target dir swap. fixed to write to `entity.status_dir` instead